### PR TITLE
Enable amd64 OSX builds (aka native Apple Silicon support)

### PR DESCRIPTION
**Pull request by mjuric**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mjuric** [comment]
Hi @hannorein et al. -- I think this should be OK to merge now.

The important changes were in `recipe/meta.yaml` (added the cross-platform stanzas, and numpy to host/run requirements), and `conda-forge.yml` (enabled the arm64 builds). All tests seem to pass -- hope it looks OK!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,17 +13,22 @@ source:
...
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
     - {{ compiler('c') }}
   host:
     - python
+    - numpy
     - pip
   run:
     - python
+    - numpy
```
Comment (mwcraig) [inline recipe/meta.yaml:31]: Should this be

```suggestion
    - {{ pin_compatible('numpy') }}
```

based on: https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

**Comment by mwcraig** [comment]
Thanks for doing this, @mjuric!

This looks good to me except for changing the runtime numpy dependency if rebound links to numpy during the compilation (I don't remember if it does).

**Comment by mjuric** [comment]
Hi @mwcraig -- you were right, numpy was unnecessary.

It was originally in the `build:` section, so I assumed it was being linked against, but grepping through the REBOUND repo I see it's only being used in tests (which are not run in the current conda build process).

I now removed it from all requirements sections in `meta.yaml` (& rebased) -- hope it looks OK (the feedstock build checks pass).

**Comment by hannorein** [comment]
I much appreciate the help on this!

**Pull request merged**